packages/domain/lib/usecases/login_validation_usecase.dart

```diff
@@ -0,0 +1,41 @@
+import 'package:domain/entities/user_entity.dart';
+import 'package:domain/entities/validation_entity.dart';
+import 'package:domain/usecases/usecase.dart';
+
+class LoginValidationUseCase
+    implements UseCaseParams<UserEntity, ValidationEntity> {
+  final loginLength = 8;
+  final passwordRegExp = r'^[a-zA-Z0-9]{7,}$';
+
+  @override
+  ValidationEntity call(UserEntity params) {
+    final loginIsEmpty = params.login.isEmpty;
+    final passwordIsEmpty = params.password.isEmpty;
+
+    if (loginIsEmpty && passwordIsEmpty) {
+      return ValidationEntity(
+        loginIsEmpty,
+        false,
+        passwordIsEmpty,
+        false,
+      );
+    }
+
+    final loginIsValid = _validateLogin(params.login);
+    final passwordIsValid = _validatePassword(params.password);
+
+    return ValidationEntity(
+      loginIsEmpty,
+      loginIsValid,
+      passwordIsEmpty,
+      passwordIsValid,
+    );
```

Comment:
need to think how we can unify it
loginIsEmpty should be a part of loginIsValid(as if it is empty then it's by default is not valid)

Reply:
Fixed.